tensorflow/python/eager/function_test.py

```diff
@@ -3931,6 +3932,83 @@ def gradients(inputs):
 
     gradients(constant_op.constant([[[1.0], [2.0]]]))  # No error is raised
 
+  def testTraceWithAnnotationsBasic(self):
```

Comment:
Be sure to include a few tests where not all args are TensorLike, to make sure they are not inadvertently converted. For example, mix Tensor with non-Tensor args, Tensor with non-Tensor named args, Tensor args with non-Tensor *args, Tensor named args with non-Tensor **kwargs, non-Tensor with Tensor *args and non-Tensor named args with Tensor **kwargs.
The list is quite long, so you'll likely want a helper: `assertTraceCount(func, 1, use_type_annos=True)`, `assertTraceCount(func, 3, use_type_annos=False)`.